src/UI/Implementation/Component/Modal/ModalHelper.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+
+namespace ILIAS\UI\Implementation\Component\Modal;
+use ILIAS\UI\Implementation\Component\Button\Primary;
+use ILIAS\UI\Implementation\Component\Button\Standard;
+
+/**
+ * @author Stefan Wanzenried <sw@studer-raimann.ch>
+ */
+trait ModalHelper
+{
+
+    /**
+     * @param string $label
+     * @return \ILIAS\UI\Component\Button\Primary|\ILIAS\UI\Component\Button\Standard
+     */
+    protected function getCancelButton($label = '')
+    {
+        $button = new Standard($label, '');
+        return $button->withOnLoadCode(function ($id) {
+            return "$('#{$id}').click(function() { $(this).closest('.modal').modal('hide'); return false; });";
```

Comment:
This getCancelButtonMethod should be discussed today for several reasons:

Does it really belong here (into the components). I do no like anything rendered at this place (including JS code)?
Would this not be better placed in the renderer? (this is actually an old discussion, where to generete UI-Elements not passed by the user)
The JS code should not be this long here. There should be something like: $('#{$id}').click(function() { callToJSLogicFromFile(this) });";